dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -120,7 +120,7 @@ internal ClientCore(ILogger? logger = null)
 
         this.CaptureUsageDetails(responseData.Usage);
         var metadata = GetResponseMetadata(responseData);
-        return responseData.Choices.Select(choice => new TextContent(choice.Text, this.DeploymentOrModelName, choice, Encoding.UTF8, metadata)).ToList();
+        return responseData.Choices.Select(choice => new TextContent(choice.Text, this.DeploymentOrModelName, choice, Encoding.UTF8, new Dictionary<string, object?>(metadata))).ToList();
```

Comment:
Why is it we need a new dictionary and can't just pass through the original?

Reply:
The metadata when is changed in an FunctionResult or a ContentBase or a StreamingContentBase, if thats not done will have the same reference, and changes to it will reflect on all chunks and choices also.

Reply:
How common will it be to have multiple Choices?  That's pretty rare, right? Might be nice to avoid the unnecessary copy if Choices.Count == 1.  This could be something like:
```C#
var results = new TextContent[responseData.Choices.Count];
for (int i = 0; i < results.Length; i++)
{
    results[i] = new TextContent(choice.Text, this.DeploymentOrModelName, choice, Encoding.UTF8,
        i == 0 ? metadata : new Dictionary<string, object?>(metadata));
}
return results;
```